extensions/cron/README.md

```diff
@@ -0,0 +1,112 @@
+# @loopback/cron
+
+This module contains a component to provide integration with
+[Cron](https://github.com/kelektiv/node-cron) so that applications can schedule
+jobs using `cron` based schedule.
+
+## Installation
+
+```sh
+npm install --save @loopback/cron
+```
+
+## Basic use
+
+### Register the Cron component
+
+The component should be loaded in the constructor of your custom Application
+class.
+
+Start by importing the component class:
+
+```ts
+import {CronComponent} from '@loopback/cron';
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(CronComponent);
+```
+
+### Register Cron jobs
+
+We can create an instance of `CronJob` and bind it to the application context so
+it can be managed by the `CronComponent` life cycle.
+
+```ts
+import {CronJob, asCronJob} from '@loopback/cron';
+
+const job = new CronJob({
+  cronTime: '*/1 * * * * *', // Every one second
+  onTick: () => {
+    // Do the work
+  },
+  runOnInit: true, // Start the job immediately
+});
+app
+  .bind('cron.jobs.job1')
+  .to(job)
+  .apply(asCronJob);
+```
+
+Alternatively, we can also define a provider class:
+
+```ts
+import {CronJob, cronJob, CronJobConfig} from '@loopback/cron';
+import {config, Provider, createBindingFromClass} from '@loopback/context';
+
+@cronJob()
+class CronJobProvider implements Provider<CronJob> {
+  constructor(@config() private jobConfig: CronJobConfig = {}) {}
+  value() {
+    const job = new CronJob({
+      // ... default config
+      ...this.jobConfig,
+    });
+    return job;
+  }
+}
+
+const jobBinding = createBindingFromClass(CronJobProvider);
+app.add(jobBinding);
+const now = new Date();
+now.setMilliseconds(now.getMilliseconds() + 10);
+const jobConfig: CronJobConfig = {
+  name: 'job-B', // Name the job
+  cronTime: sendAt(now), // Start the job in 10 ms
+  runOnInit: false,
+};
+app.configure(jobBinding.key).to(jobConfig);
+```
+
+By default, all cron jobs will be started and stopped by `CronComponent` when
+the application is started or stopped.
+
+To start a cron job after application starts, we can either set `runOnInit` to
```

Comment:
Are we really `starting` the cron job as soon as application starts? e.g. running the job. Or do we mean `register a cron job with the operating system` as soon as the app starts?

Reply:
Is the module we are using actually going to enter a cron job into linux OS? Or does it `mimic` the idea of `cron` on linux?

Reply:
It mimics the `cron` scheduler.